### DISCUSSION TITLE
Plugin to change the home page

### DISCUSSION THREAD

**Discussion by gpsblues** [GetPublii/Publii#1768] [Help]
I made a plugin that modifies some aspects of the home. The problem is to identify the home and I did this:

main.js

```
const homeUrl = rendererInstance.globalContext.website.baseUrl;
const pageUrl = rendererInstance.globalContext.website.pageUrl.replace(/\\/g,"/"); 
if (homeUrl == pageUrl) { //home page }
```
This method of string comparison works (the replace is for compatibility with windows systems), but it looks unprofessional.
Is there a better method?

Thanks a lot

**Comment by dziudek** [comment]
Hi,

Under `rendererInstance.siteConfig.advanced.pageAsFrontpage` you have ID of the page used as frontpage

You can also check `rendererInstance.siteConfig.advanced.usePageAsFrontpage` before checking if page is used as frontpage.

**Comment by gpsblues** [comment]
Thank you so much, @dziudek!

Both methods work correctly, but unfortunately, they don’t identify the site’s homepage, which could be generated by either `index.hbs` or `page.hbs`. 

When I set a page as the homepage, I get:
- `rendererInstance.siteConfig.advanced.usePageAsFrontpage = true`
- `rendererInstance.siteConfig.advanced.pageAsFrontpage = 9` (page ID).

If I then set `index` as the homepage, I get:
- `rendererInstance.siteConfig.advanced.usePageAsFrontpage = false`
- `rendererInstance.siteConfig.advanced.pageAsFrontpage = 9` (it retains the page ID, but this page is no longer the index). 

Ecco la traduzione in inglese:

I also tried using `context`, but I only get `index` if I use `index.hbs` as the frontpage.

When distributing a plugin that modifies the frontpage, I can’t know the user’s settings.

**Comment by gpsblues** [comment]
Hi,

Actually, I can determine the settings:

- `rendererInstance.siteConfig.advanced.usePageAsFrontpage = true && rendererInstance.siteConfig.advanced.pageAsFrontpage = N`: Page N is set as the homepage.
- `rendererInstance.siteConfig.advanced.usePageAsFrontpage = false`: index.hbs is the homepage.

But to make changes only on the frontpage, the only way that comes to mind right now is by comparing the URLs:

```javascript
const homeUrl = rendererInstance.globalContext.website.baseUrl;
const pageUrl = rendererInstance.globalContext.website.pageUrl.replace(/\\/g,"/");
if (homeUrl === pageUrl) {
    // homepage
}
```
